File: 프로그래머스/lv1/12901. 2016년/2016년.py
def solution(a, b):
    days_dict = {1:30, 2:29, 3:31, 4:30, 5:31, 6:30, 7:31, 8:31, 9:30, 10:31, 11:30, 12:31}
    day_list = ["FRI", "SAT", "SUN", "MON", "TUE", "WED", "THU"]
    days = b if a != 1 else b - 1
    for m in range(1, a):
        days += days_dict[m]
    
    return day_list[days % 7]

File: 프로그래머스/lv1/12901. 2016년/2016년.java
import java.util.*;

class Solution {
    public String solution(int a, int b) {
        String answer = "";
        int days = -1;
        Map<Integer, Integer> map = new HashMap<>() {{
            put(1, 31);
            put(2, 29);
            put(3, 31);
            put(4, 30);
            put(5, 31);
            put(6, 30);
            put(7, 31);
            put(8, 31);
            put(9, 30);
            put(10, 31);
            put(11, 30);
            put(12, 31);
        }};
        
        String[] dates = {"FRI","SAT","SUN","MON","TUE","WED","THU"};
        
        for (int i = 1; i < a; i++) {
            days += map.get(i);
        }
        days += b;
        
        return dates[days % 7];
    }
}
